pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -177,19 +177,50 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
 	if err != nil {
 		return fmt.Errorf("could not get a list of unhealthy members: %v", err)
 	}
+
+	// map machines pending deletion to node internal ip
+	votingMachinesPendingDeletionIndex := ceohelpers.IndexMachinesByNodeInternalIP(votingMachinesPendingDeletion)
+	// find unhealthy machine pending deletion to remove first from the cluster, without violating quorum
+	var unhealthyMachinePendingDeletion *machinev1beta1.Machine = nil
 	if len(unhealthyMembers) > 0 {
 		var unhealthyMembersURLs []string
 		for _, unhealthyMember := range unhealthyMembers {
+			unhealthyMemberNodeInternalIP, err := ceohelpers.MemberToNodeInternalIP(unhealthyMember)
+			if err != nil {
+				return fmt.Errorf("cannot get node internal ip for unhealthy member %v: %w", unhealthyMember, err)
+			}
+			unhealthyMemberMachine, ok := votingMachinesPendingDeletionIndex[unhealthyMemberNodeInternalIP]
+			if ok {
+				unhealthyMachinePendingDeletion = unhealthyMemberMachine
+			}
 			if len(unhealthyMember.PeerURLs) > 0 {
 				unhealthyMembersURLs = append(unhealthyMembersURLs, unhealthyMember.PeerURLs[0])
 			} else {
 				unhealthyMembersURLs = append(unhealthyMembersURLs, unhealthyMember.Name)
 			}
 		}
-		return fmt.Errorf("cannot proceed wth scaling down, unhealthy members found: %v", unhealthyMembersURLs)
+		if unhealthyMachinePendingDeletion != nil {
+			klog.V(4).Infof("found unhealthy voting member's machine pending deletion: %v, attempt to scale this member down since it does not violate quorum", unhealthyMachinePendingDeletion)
+			klog.V(4).Infof("unhealthy members found: %v", unhealthyMembersURLs)
+		} else {
+			return fmt.Errorf("cannot proceed wth scaling down, unhealthy members found: %v", unhealthyMembersURLs)
+		}
 	}
 
+	// trying to remove the unhealthy machine pending deletion first
```

Comment:
gr8 question, 

so there could be many machines pending deletions, however, we would need to remove the `unhealthy` members first. Otherwise, we could remove a healthy member pending deletion, while there is an `unhealthy` member pending deletion 

please correct me if i am wrong

Reply:
still thinking about it, you might want to simplify the logic by just putting the unhealthy member at the start of the `votingMachinesPendingDeletion` slice. That way we don't duplicate the deletion logic. It will also stop on the first deletion.

Reply:
I agree with you, I will try to fix it now 👍🏽

Reply:
@tjungblu  fixed :) .. can u have a look now please ?

Reply:
much better! thank you sir